.github/workflows/build-cross.yml

```diff
@@ -0,0 +1,293 @@
+      - name: Upload Artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: windows-executable
+          path: example.exe
+          retention-days: 1
+      - name: Test cross sak
+        run: |
+          printf %s "$STR_UTF16" > utf16.ref
+          printf %s "$STR_UTF8" > utf8.ref
+          set -x
+          runtime/sak.exe encode-C-utf16-literal "$TESTDIR" > utf16
+          git diff --no-index utf16.ref utf16
+          runtime/sak.exe encode-C-utf8-literal "$TESTDIR" > utf8
+          git diff --no-index utf8.ref utf8
+
+  run-windows:
+    runs-on: windows-latest
+    needs: cross-windows
+    steps:
+      - name: Download Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: windows-executable
+      - name: Run example program
+        run: |
+          .\example.exe
+      - name: Set up MSVC
+        uses: ilammy/msvc-dev-cmd@v1
+        with:
+          arch: x64
+      - name: Fetch OCaml
+        uses: actions/checkout@v4
+      - name: Generate a dummy misc.h and result files to test sak
+        shell: bash
+        run: |
+          echo '#include <stdlib.h>' > runtime/caml/misc.h
+          echo 'typedef wchar_t char_os;' >> runtime/caml/misc.h
+          echo '#define T(x) L ## x' >> runtime/caml/misc.h
+          echo '#define main_os wmain' >> runtime/caml/misc.h
+          echo '#define strcmp_os wcscmp' >> runtime/caml/misc.h
+          printf %s "$STR_UTF16" > utf16.ref
+          printf %s "$STR_UTF8_ENC" > utf8.ref
+      - name: Build sak
+        run: >
+          cl -std:c11 -experimental:c11atomics -I runtime
+          -D__USE_MINGW_ANSI_STDIO=0 -DUNICODE -D_UNICODE -DWINDOWS_UNICODE=1
+          /Fesak.exe runtime\sak.c
+      - name: Test cross sak
+        run: |
+          .\sak.exe encode-C-utf16-literal "${env:TESTDIR}" > utf16
+          git diff --no-index utf16.ref utf16
+          .\sak.exe encode-C-utf8-literal "${env:TESTDIR}" > utf8
+          git diff --no-index utf8.ref utf8
+
+  cross-arm-linux:
+    runs-on: ubuntu-latest
+    needs: non-cross
+    steps:
+      - name: Download Artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: non-cross-ocaml
+      - name: Install non-cross OCaml and set up environment
+        run: |
+          set -x
+          tar xaf ocaml.tar.zst -C "$HOME"
+          rm -f ocaml.tar.zst
+          echo "$HOME/.local/bin" >> "$GITHUB_PATH"
+          sudo apt-get install -y gcc-aarch64-linux-gnu qemu-user
+      - name: Checkout OCaml
+        uses: actions/checkout@v4
+        with:
+          submodules: true
+          persist-credentials: false
+      - name: Configure, build and install Linux-to-Windows OCaml
+        run: |
+          set -x
+          if ! ./configure --prefix="$HOME/cross" --target=aarch64-linux-gnu ;
+          then
+            cat config.log
+            exit 1
+          fi
```

Comment:
```suggestion
          ./configure --prefix="$HOME/cross" --target=aarch64-linux-gnu || failed=$?
          if ((failed)) ; then
            cat config.log
            exit $failed
          fi
```

Reply:
As I don’t intend the CI workflow to be merged (at least as is), I’ll ignore your style remark there for now.